### ISSUE TITLE
Spell checking tweak for ignoring pandoc references?

### ISSUE THREAD

**Issue by paultopia** [jrblevin/markdown-mode#572]
Hi there, 

Is there any interest in adding pandoc-style reference links (of the form `[@refmarker]`) to the tokens ignored by flyspell?  

Right now, I've just implemented this as an advice to markdown-mode in my own init with the following code, but it seems like it might be useful to others, and am happy to turn it into a contribution to this package if there's interest.

```elisp
  (defun is-ampersand (s)
  (string= "@" s))

  (defun not-cite ()
   (save-excursion
    (forward-word -1)
    (let ((result (is-ampersand (string (preceding-char)))))
     (not result))))

(defun not-cite-advice (orig &rest args)
(if (not-cite)
(apply orig args)))

  (advice-add 'markdown-flyspell-check-word-p :around #'not-cite-advice)
```

**Comment by atanasj** [comment]
This would be a welcome feature and I have tried to implement this, but it works inconsistently in that it `flyspell` still identifies certain references as a spelling error. Not sure what I am doing wrong... Any additional advice @paultopia?

**Comment by syohex** [comment]
I'm not sure about pandoc-style. Could you show us test cases(or sample texts) of this ?

**Comment by atanasj** [comment]
A sample text would be something line this:

```markdown
The number of subgroups in our findings are consistent with recent
investigations into latent profiles of suicidal individuals that tend to find
three groups using varying on factors associated with suicide [@wong2020;
@weintraub2020; @bertuccio2021; @love2021; @xiao2021].
```

Which looks like: 

![image](https://user-images.githubusercontent.com/27214625/171559508-b6b3be68-02c0-4a00-b9ec-8bcf1cc00192.png)

There should be no identified spelling errors for `pandoc` style citations, e.g., `@refyear`.

Not sure if that helps.

**Comment by syohex** [comment]
I have merged #710. Please check lates version.
